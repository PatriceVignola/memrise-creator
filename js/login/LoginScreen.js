/* @flow */

import React from 'react';
import { Button } from 'react-native';
import {
  type NavigationScreenProp,
  NavigationActions,
} from 'react-navigation';
import loginWithGoogle from '../actions/login';

type Props = {
  navigation: NavigationScreenProp<any>,
};

export default class LoginScreen extends React.Component<Props> {
  constructor(props: Props) {
    super(props);

    this.onGoogleLoginPress = this.onGoogleLoginPress.bind(this);
  }

  onGoogleLoginPress = () => {
    this.props.navigation.dispatch(NavigationActions.reset({
      index: 0,
      actions: [NavigationActions.navigate({ routeName: 'GoogleLogin' })],
    }));

    loginWithGoogle();
  }

  render() {
    return (
      // TODO: Add the google logo
      // TODO: Style the button according to the guidelines
      // https://developers.google.com/identity/branding-guidelines#g+signin-social-scopes
      <Button onPress={this.onGoogleLoginPress} title="Sign in with Google" />
    );
  }
}
