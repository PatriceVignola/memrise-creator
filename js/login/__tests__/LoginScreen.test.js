/* @flow */

import React from 'react';
import { addNavigationHelpers } from 'react-navigation';
import ShallowRenderer from 'react-test-renderer/shallow';
import LoginScreen from '../LoginScreen';

const renderer = new ShallowRenderer();

test('Matches snapshot', () => {
  const navigation = addNavigationHelpers({
    state: {},
    dispatch: jest.fn(),
  });

  const result = renderer.render((
    <LoginScreen
      navigation={navigation}
    />
  ));
  expect(result).toMatchSnapshot();
});
