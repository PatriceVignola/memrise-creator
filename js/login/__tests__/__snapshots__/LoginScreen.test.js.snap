// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<Button
  onPress={[Function]}
  title="Sign in with Google"
/>
`;
